Source/WebCore/platform/graphics/gstreamer/mse/PlaybackPipeline.cpp

```diff
@@ -347,10 +347,8 @@ void PlaybackPipeline::flush(AtomString trackId)
     GST_DEBUG_OBJECT(appsrc, "segment: [%" GST_TIME_FORMAT ", %" GST_TIME_FORMAT "], rate: %f",
         GST_TIME_ARGS(start), GST_TIME_ARGS(stop), rate);
 
-    if (!gst_element_send_event(GST_ELEMENT(appsrc), gst_event_new_flush_start())) {
+    if (!gst_element_send_event(GST_ELEMENT(appsrc), gst_event_new_flush_start()))
         GST_WARNING("Failed to send flush-start event for trackId=%s", trackId.string().utf8().data());
-        return;
-    }
```

Comment:
When pipeline is not ready yet, sending flash-start and flash-stop events may return false (not sure why this happens). But, we still need to flash appsrc, so I removed returns for this cases.

Reply:
The above hang is caused by a blocking pad probe installed by playbin. Plabin blocks input-selector src pad until all streams are known. the AbortVP9Video test case doesn't append data to the audio source buffer, so audio track is not configured at the time when video is flushing. 

The reproduction is random because sometimes test triggers abort before any data is provided to the video pipeline. In which case flushing was not triggered and the test suceeded.

I've added a check that will skip flushing if not all tracks are configured yet.

Reply:
> reproduction rate is 9 / 10 for me.
9 failures for every 10 test runs or 9 passes of every 10 test runs?

> Plabin blocks input-selector src pad until all streams are known.
This is correct. That probe (combine->block_id, block_serialized_data_cb) is removed on no-more-pads. It should still allow for flush events to flow though.

> I suspect similar might be reproducible with upstream version. do you see same issue there?
This should not be an issue upstream. The reworked WebKitMediaSrc, by design, doesn't start pushing any buffers until no-more-pads is emitted.

Reply:
> 9 failures for every 10 test runs or 9 passes of every 10 test runs?

it is 9 passes and 1 faliure. I should have been more clear. Sorry for the confusion.